src/Gui/NaviCube.cpp

```diff
@@ -263,6 +266,31 @@ void NaviCube::setCorner(Corner c) {
     m_NaviCubeImplementation->m_PrevHeight = 0;
 }
 
+void NaviCube::setSize(int size)
+{
+    m_NaviCubeImplementation->m_CubeWidgetSize = size;
```

Comment:
Having a setter for this property is cumbersome. On the one hand the method can be used to set a value but on the other hand the NaviCubeImplementation retrieves the value directly from the parameter group. Since now we have mixed responsibility it is not clear who sets values which will cause some undesired side-effects.

IMO, the NaviCubeImplementation class shouldn't access the parameter group directly but provide member variables and the corresponding setter methods. It will be up to the View3DInventor class to call the setter methods.

Reply:
OK. I improved the situation now a bit. I will not go further in the PR since this PR is meant to fix the missing handling of the NaviCube parameters.

I have not written the NaviCube feature. It was obviously not well maintained. In my previous merged PR, I deleted for example a year old class that was never used. In the NaviCube code i see further things like this: FontStretch, OffsetX settings etc. IMO these can be removed since they are nowhere set, except an expert user sets them.